.github/workflows/ci.yml

```diff
@@ -116,6 +120,90 @@ jobs:
           secret_vault_address: ${{ secrets.VAULT_ADDR }}
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
 
+  docker-checks:
+    if: needs.detect-changes.outputs.docker-ci == 'true'
+    needs: [ detect-changes ]
+    runs-on: ubuntu-latest
```

Comment:
Please add a `timeout-minutes` variable to limit execution time to what is conservatively needed based on observation how long it usually runs.

Reply:
the first successful run took ~15 minutes https://github.com/camunda/camunda/actions/runs/9664124006/job/26658087425?pr=19623
I will put 25 minutes timeout for now, which less than the timeout of the cocurrent job `java-unit-tests` (30 min)

Reply:
The `java-unit-tests` job also doesn't need the 30 minutes, it is closer to 15 the last time I checked.

I also want to keep runtimes low so 15 minutes is already at the upper bound (I put some new guidelines in https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria ) - we have to make sure to not let it grow, otherwise parallelization/caching/bigger runners need to be considered.

Reply:
ok, I think 20 minutes is a fair value.
I feel like I am negotiating the price of my car 😆

Reply:
@cmur2 
I completely agree that capping job execution time to 15 minutes is a good goal. Currently, I've set the limit to 20 minutes to match the duration of the integration-tests job, which is our longest-running task. However, I'm concerned that the job duration will increase as we add new modules to the project (such as identity, optimize, etc.).
I believe there are optimizations we can implement in the frontend build, which is currently done sequentially due to a known issue with the frontend-maven-plugin that breaks in parallel build. There's an initiative underway to consolidate the frontend code of all components. This will allow us to implement further optimizations in the frontend build process.
<img width="1255" alt="image" src="https://github.com/camunda/camunda/assets/146710021/f6d8b3ad-98c9-412d-a5ab-3fb739b5e70f">

<img width="1322" alt="image" src="https://github.com/camunda/camunda/assets/146710021/c66c3ea3-8bdb-4974-ae6a-d18861dbe73c">

Reply:
I agree with the current settings.

> that the job duration will increase

This will most likely happen but it is also the reason why I set a goal: adding new modules needs to be counterbalanced by investing into CI improvements to avoid runtime creep over the coming months and the goal helps us decide when is the time for what - it is unsustainable to only ever add code without investing into CI.